### PR TITLE
FPS-321 - added api reference pages for FPS swagger

### PR DESCRIPTION
**Pull request by rahulprabhune-macrometa**
FPS-321 - added api reference pages for FPS swagger

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -9,7 +9,7 @@
     Macrometa is a secure, global data platform with integrated pub/sub, stream processing, search, functions, and containers. Create stateful APIs and real-time, event-driven, streaming data apps in minutes – not months.
     <br />
     <br />
-    <a href="https://www.macrometa.com/docs">Explore the Docs 🚀</a>
+    <a href="https://www.macrometa.com/docs">Explore thee Docs 🚀</a>
```
Comment (rahulprabhune-macrometa) [inline README.md]:
```suggestion
    <a href="https://www.macrometa.com/docs">Explore the Docs 🚀</a>
```

**Review by joubert-macrometa — APPROVED** [review]

File: src/pages/apiVwrsService.js
Comment (diana-macrometa) [inline src/pages/apiVwrsService.js:1]: Do we really want to delete the apiVwrsService page? Won't that kill the VWRS API page?

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 2]
```diff
@@ -0,0 +1,870 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "PhotonIQ Digital Fingerprint Service API",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
    "title": "PhotonIQ Digital Fingerprinting API",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 3]
```diff
@@ -0,0 +1,870 @@
...
+                "properties": {
+                  "matchScore": {
+                    "type": "number"
+                  }
+                }
+              },
+              "timestamp": {
+                "type": "number",
+                "format": "date-time",
+                "description": "Visit timestamp"
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: All descriptions should end with a period. Summaries should not have periods at the end.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 4]
```diff
@@ -0,0 +1,870 @@
...
+              },
+              "firstSeenAt": {
+                "type": "number",
+                "format": "date-time",
+                "description": "Visitor createdAt"
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: Not sure what you mean by "createdAt" here. What is the user supposed to understand from this description?

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 5]
```diff
@@ -0,0 +1,870 @@
...
+        "security": [
+          {
+            "ApiKeyAuth": []
+          }
+        ],
+        "description": "Get the version details for the service.",
+        "operationId": "getVersion",
+        "responses": {
+          "200": {
+            "description": "Successful returned the version.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "Successfully returned the version.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 6]
```diff
@@ -0,0 +1,870 @@
...
+                "$ref": "#/components/schemas/RecordVisitInformation"
+              }
+            }
+          }
+        },
+        "parameters": [
+          {
+            "in": "header",
+            "name": "Cookie",
+            "description": "If this cookie is present it means that we have seen this visitor before.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "If this cookie is present, then this is a returning visitor.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 7]
```diff
@@ -0,0 +1,870 @@
...
+        ],
+        "responses": {
+          "202": {
+            "description": "Successfully recorded the visit.",
+            "headers": {
+              "x-photoniq-dsid": {
+                "schema": {
+                  "type": "string"
+                },
+                "description": "Digital Signature ID.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
                "description": "Digital signature ID.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 8]
```diff
@@ -0,0 +1,870 @@
...
+                "example": 12416155251620899000
+              },
+              "x-photoniq-vid": {
+                "schema": {
+                  "type": "string"
+                },
+                "description": "The visitor ID.",
+                "example": "062b8104-6a71-4ac3-ac7f-6cd1ba6f2fa1"
+              },
+              "Cookie": {
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json:468]: Why is this uppercase?

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 9]
```diff
@@ -0,0 +1,870 @@
...
+            "description": "Bad request."
+          },
+          "401": {
+            "description": "Authorization failure due to invalid authentication credentials."
+          }
+        }
+      },
+      "get": {
+        "summary": "Retrieve data for a particular visit",
+        "description": "Returns data for a particular visit based on the visitId.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
        "description": "Returns data for a particular visit based on the visit ID.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 10]
```diff
@@ -0,0 +1,870 @@
...
+          "401": {
+            "description": "Authorization failure due to invalid authentication credentials."
+          }
+        }
+      }
+    },
+    "/api/ds/v1/metrics": {
+      "get": {
+        "summary": "Retrieve metrics data",
+        "description": "Returns metrics data for a given time period. If the data for a particular period has a value of 'null', it denotes that the metrics could not be calculcated for that time period. For example accuracy cannot be calculated if there was no cookied visit for that day.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
        "description": "Returns metrics data for a given time period. If the data for a particular period has a value of 'null', then metrics could not be calculated for that time period. For example, accuracy cannot be calculated if there was no visits for that day.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 11]
```diff
@@ -0,0 +1,870 @@
...
+        "security": [
+          {
+            "ApiKeyAuth": []
+          }
+        ],
+        "parameters": [
+          {
+            "name": "n",
+            "in": "query",
+            "description": "Number of top visits data to return to.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: I do not know what this description means.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 12]
```diff
@@ -0,0 +1,870 @@
...
+            "required": false,
+            "example": "false",
+            "schema": {
+              "type": "boolean"
+            }
+          },
+          {
+            "name": "start",
+            "in": "query",
+            "description": "The start date used for querying visits data. Must be provided in RFC3339 date-time format and be in UTC. The default start time is 24 hours from current time.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "The start date used for querying visits data. Must be provided in RFC3339 date-time format and be in UTC. The default start time is 24 hours before the current time.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 13]
```diff
@@ -0,0 +1,870 @@
...
+        "security": [
+          {
+            "ApiKeyAuth": []
+          }
+        ],
+        "parameters": [
+          {
+            "name": "start",
+            "in": "query",
+            "description": "The start date used for querying visitor traffic data. Must be provided in RFC3339 date-time format and UTC. The default start time is 24 hours from current time.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "The start date used for querying visitor traffic data. Must be provided in RFC3339 date-time format and UTC. The default start time is 24 hours before the current time.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 14]
```diff
@@ -0,0 +1,870 @@
...
+        }
+      }
+    },
+    "/api/ds/v1/telemetry": {
+      "get": {
+        "summary": "Provides telemetry data",
+        "tags": [
+          "Telemetry"
+        ],
+        "description": "Provide telemetry data in Prometheus format. Useful for monitoring services to monitor the health of the Digital Fingerprint service.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
        "description": "Provide telemetry data in Prometheus format. Useful for monitoring services to monitor the health of the Digital Fingerprinting service.",
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

**Comment by joubert-macrometa** [comment]
@rahulprabhune-macrometa remember that these json changes need to also be brought into the original swagger file in our fps repo.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 15]
```diff
@@ -0,0 +1,870 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "PhotonIQ Digital Fingerprinting API",
+    "description": "API documentation for the PhotonIQ Digital Fingerprint Service",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
    "description": "API documentation for the PhotonIQ Digital Fingerprinting Service",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 16]
```diff
@@ -0,0 +1,870 @@
...
+                "properties": {
+                  "matchScore": {
+                    "type": "number"
+                  }
+                }
+              },
+              "timestamp": {
+                "type": "number",
+                "format": "date-time",
+                "description": "Visit timestamp"
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
                "description": "Timestamp of the visit."
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 4 above)
Comment (diana-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: @rahulprabhune-macrometa and @joubert-macrometa, please answer the question.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 17]
```diff
@@ -0,0 +1,870 @@
...
+                "$ref": "#/components/schemas/RecordVisitInformation"
+              }
+            }
+          }
+        },
+        "parameters": [
+          {
+            "in": "header",
+            "name": "Cookie",
+            "description": "If this cookie is present then this is a returning visitor before.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "If this cookie is present, then this is a returning visitor.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 18]
```diff
@@ -0,0 +1,870 @@
...
+              },
+              "x-photoniq-vid": {
+                "schema": {
+                  "type": "string"
+                },
+                "description": "The visitor ID.",
+                "example": "062b8104-6a71-4ac3-ac7f-6cd1ba6f2fa1"
+              },
+              "Cookie": {
+                "description": "Security features will be enabled for this cookie to minimize tampering.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json:469]: "will be" or "are"?

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 19]
```diff
@@ -0,0 +1,870 @@
...
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/VisitDetails"
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request"
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "Bad request."
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 20]
```diff
@@ -0,0 +1,870 @@
...
+            "description": "Visitor ID.",
+            "required": true,
+            "schema": {
+              "type": "string"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successful retrieved visitor data.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "Successfully retrieved visitor data.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 21]
```diff
@@ -0,0 +1,870 @@
...
+            "content": {
+              "application/json": {
+                "schema": {
+                  "$ref": "#/components/schemas/AllVisitsResponse"
+                }
+              }
+            }
+          },
+          "400": {
+            "description": "Bad request"
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "Bad request."
```

Comment (diana-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: @joubert-macrometa , isn't there a longer version of this? Something about missing or malformed data?

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 11 above)
Comment (diana-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: @joubert-macrometa and @rahulprabhune-macrometa, please explain this.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 22]
```diff
@@ -0,0 +1,870 @@
...
+        "security": [
+          {
+            "ApiKeyAuth": []
+          }
+        ],
+        "parameters": [
+          {
+            "name": "start",
+            "in": "query",
+            "description": "The start date used for querying the device data. Must be provided in RFC3339 date-time format and UTC. The default start time is 24 hours from current time.",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "The start date used for querying the device data. Must be provided in RFC3339 date-time format and UTC. The default start time is 24 hours from the current time.",
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 23]
```diff
@@ -0,0 +1,870 @@
...
+                  "example": "# HELP req_metric Metric updated every minute for total request in that minute\n# TYPE req_metric gauge\nreq_metric{endTimestamp=\"1695930720000\",metricName=\"requestCount\",startTimestamp=\"1695930660000\"} 0\n# HELP req_ps_metric Metric updated every minute for request per second\n# TYPE req_ps_metric gauge\nreq_ps_metric{endTimestamp=\"1695930720000\",metricName=\"requestCountPerSecond\",startTimestamp=\"1695930660000\"} 0\n"
+                }
+              }
+            }
+          },
+          "401": {
+            "description": "Authorization failure due to invalid authentication credentials."
+          },
+          "403": {
+            "description": "Either the API doesn't have permissions or it is deactivated."
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json:864]: Do you mean API key here?

**Review by diana-macrometa — CHANGES_REQUESTED** [review]
@rahulprabhune-macrometa and @joubert-macrometa, still a bunch of questions and edits.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 18 above)
Comment (rahulprabhune-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json:469]: will be

File: src/pages/apiVwrsService.js
Comment (rahulprabhune-macrometa) [reply src/pages/apiVwrsService.js:1]: that was @rahul-naik1 's suggestion

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json [hunk 24]
```diff
@@ -0,0 +1,870 @@
+{
+  "openapi": "3.0.0",
+  "info": {
+    "title": "PhotonIQ Digital Fingerprinting API",
+    "description": "API documentation for the PhotonIQ Digital Fingerprinting Service",
```
Comment (diana-macrometa) [inline lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
    "description": "API documentation for the PhotonIQ Digital Fingerprinting",
```

**Review by diana-macrometa — CHANGES_REQUESTED** [review]

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 4 above)
Comment (joubert-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
                "description": "Timestamp of the first time the visitor was seen."
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 11 above)
Comment (joubert-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]:
```suggestion
            "description": "Number of visitors.",
```

Comment (joubert-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: @diana-macrometa I think this is better.  But I am not sure that does the job.  

The api returns the top N number of visitors.  This parameter is what you put in for N.  I could not come up with a better way of saying this.

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 23 above)
Comment (joubert-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json:864]:
```suggestion
            "description": "Access to this API is either restricted due to insufficient permissions, or the API key is no longer active.."
```

File: lib/openapi/revisions/fps-spec-1.5.0-20231130.json (hunk 11 above)
Comment (diana-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: Is it the maximum number of visitors in a given time window?

Comment (rahulprabhune-macrometa) [reply lib/openapi/revisions/fps-spec-1.5.0-20231130.json]: This is visitors with maximum visits in the given time window

**Review by diana-macrometa — APPROVED** [review]

**Comment by diana-macrometa** [comment]
Any idea when this will be merged? It's over a month old.

**Pull request merged**